### PR TITLE
SUS-1090: Refactor Phalanx page move handling

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-1090
- All page move related hooks moved to `PhalanxContentBlock`
- `SpecialMovepageBeforeMove` handler is now responsible for Special:MovePage and checks title and summary
- `AbortMove` handler is now responsible for page moves via API and checks title and summary
- removed legacy MediaWiki:Spam-whitelist support

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 25 issues
* MAJOR 9 major
* MINOR 13 minor
* INFO 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PhalanxModel.php#L50](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/models/PhalanxModel.php#L50): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [PhalanxModel.php#L75](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/models/PhalanxModel.php#L75): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. MAJOR [PhalanxModel.php#L106](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/models/PhalanxModel.php#L106): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1067)
1. MAJOR [PhalanxModel.php#L117](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/models/PhalanxModel.php#L117): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): The Cyclomatic Complexity of this function "sendToPhalanxDaemon" is 28 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): This function "sendToPhalanxDaemon" has 155 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)
1. MAJOR [PhalanxService.class.php#L204](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L204): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. MAJOR [PhalanxService.class.php#L278](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L278): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
1. MAJOR [PhalanxService.class.php#L285](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L285): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. MINOR [Phalanx_setup.php#L129](https://github.com/Wikia/app/blob/2521ed8d1bc84408a8a99640b369017cbcb69793/extensions/wikia/PhalanxII/Phalanx_setup.php#L129): Define a constant instead of duplicating this literal "phalanx" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 25 issues
* MAJOR 9 major
* MINOR 13 minor
* INFO 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PhalanxModel.php#L50](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/models/PhalanxModel.php#L50): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [PhalanxModel.php#L75](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/models/PhalanxModel.php#L75): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. MAJOR [PhalanxModel.php#L106](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/models/PhalanxModel.php#L106): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1067)
1. MAJOR [PhalanxModel.php#L117](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/models/PhalanxModel.php#L117): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): The Cyclomatic Complexity of this function "sendToPhalanxDaemon" is 28 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): This function "sendToPhalanxDaemon" has 155 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)
1. MAJOR [PhalanxService.class.php#L204](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L204): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. MAJOR [PhalanxService.class.php#L278](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L278): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
1. MAJOR [PhalanxService.class.php#L285](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L285): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. MINOR [Phalanx_setup.php#L129](https://github.com/Wikia/app/blob/dc7e598c65201369578dfe538e71d8f0de91ebbe/extensions/wikia/PhalanxII/Phalanx_setup.php#L129): Define a constant instead of duplicating this literal "phalanx" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/PhalanxII/models/PhalanxModel.php [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <?php
 
 /**
- * @method setBlock
+ * @method setBlock( $block )
  * @method getBlock
- * @method setText
+ * @method setText( string $text )
```
Comment (macbre) [inline extensions/wikia/PhalanxII/models/PhalanxModel.php]: :+1:

File: extensions/wikia/PhalanxII/hooks/PhalanxContentBlock.class.php [hunk 2]
```diff
@@ -102,33 +102,70 @@ static private function checkContentFromEditPage( $editPage, $errorFunctionName
...
-	 * Aborts a page move if the summary given matches
-	 * any blacklisted phrase.
+	 * Handles page moves via Special:MovePage - aborts if the summary given matches any blacklisted phrase,
+	 * or if the destination title matches any title filter.
+	 *
+	 * @param MovePageForm $move Special:MovePage class instance
+	 * @return bool False if a match was found to stop hook processing, true otherwise
 	 */
-	static public function abortMove( $oldTitle, $newTitle, $user, &$error, $reason ) {
+	static public function beforeMove( MovePageForm $move ): bool {
```
Comment (macbre) [inline extensions/wikia/PhalanxII/hooks/PhalanxContentBlock.class.php:115]: Yay for using PHP 7 features

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 25 issues
* MAJOR 9 major
* MINOR 13 minor
* INFO 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PhalanxModel.php#L50](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/models/PhalanxModel.php#L50): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [PhalanxModel.php#L75](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/models/PhalanxModel.php#L75): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. MAJOR [PhalanxModel.php#L106](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/models/PhalanxModel.php#L106): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1067)
1. MAJOR [PhalanxModel.php#L117](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/models/PhalanxModel.php#L117): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): The Cyclomatic Complexity of this function "sendToPhalanxDaemon" is 28 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): This function "sendToPhalanxDaemon" has 155 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)
1. MAJOR [PhalanxService.class.php#L204](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L204): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. MAJOR [PhalanxService.class.php#L278](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L278): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
1. MAJOR [PhalanxService.class.php#L285](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L285): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. MINOR [Phalanx_setup.php#L129](https://github.com/Wikia/app/blob/86a73961f63e45a9017ba2f6e3f8ce32eaedd514/extensions/wikia/PhalanxII/Phalanx_setup.php#L129): Define a constant instead of duplicating this literal "phalanx" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/PhalanxII/models/PhalanxContentModel.php [hunk 3]
```diff
@@ -45,7 +45,8 @@ public function displayBlock() {
 			$this->wg->Out->addWikiMsg( 'phalanx-content-spam-summary' );
 		}
 
-		$this->wg->Out->returnToMain( false, $this->title );
+		// SUS-1090: Use page title, not destination page
+		$this->wg->Out->returnToMain( false, $this->wg->Title );
```
Comment (macbre) [inline extensions/wikia/PhalanxII/models/PhalanxContentModel.php:49]: :+1:

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 25 issues
- MAJOR 9 major
- MINOR 13 minor
- INFO 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [PhalanxModel.php#L50](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/models/PhalanxModel.php#L50): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
2. MAJOR [PhalanxModel.php#L75](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/models/PhalanxModel.php#L75): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
3. MAJOR [PhalanxModel.php#L106](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/models/PhalanxModel.php#L106): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1067)
4. MAJOR [PhalanxModel.php#L117](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/models/PhalanxModel.php#L117): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
5. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): The Cyclomatic Complexity of this function "sendToPhalanxDaemon" is 28 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
6. MAJOR [PhalanxService.class.php#L154](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L154): This function "sendToPhalanxDaemon" has 155 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)
7. MAJOR [PhalanxService.class.php#L204](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L204): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
8. MAJOR [PhalanxService.class.php#L278](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L278): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
9. MAJOR [PhalanxService.class.php#L285](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/services/PhalanxService.class.php#L285): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
10. MINOR [Phalanx_setup.php#L129](https://github.com/Wikia/app/blob/1e6bfa0dd6959ab16d9f3e5bcf9fc10c75f4dab6/extensions/wikia/PhalanxII/Phalanx_setup.php#L129): Define a constant instead of duplicating this literal "phalanx" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by bulaszenko** [comment]
Checked works good :)

**Pull request merged**